pkg/virt-api/rest/subresource.go

```diff
@@ -598,13 +598,19 @@ func (app *SubresourceAPIApp) StopVMRequestHandler(request *restful.Request, res
 		if !errors.IsNotFound(err) {
 			writeError(errors.NewInternalError(err), response)
 			return
-		} else {
-			writeError(errors.NewConflict(v1.Resource("virtualmachine"), name, fmt.Errorf("VM is not running")), response)
-			return
 		}
+
+		writeError(errors.NewConflict(v1.Resource("virtualmachine"), name, fmt.Errorf("VM has no associated VMI running")), response)
+		return
 	}
-	if vmi == nil || vmi.IsFinal() || vmi.Status.Phase == v1.Unknown || vmi.Status.Phase == v1.VmPhaseUnset {
-		writeError(errors.NewConflict(v1.Resource("virtualmachine"), name, fmt.Errorf("VM is not running")), response)
+
+	if vmi == nil || vmi.Status.Phase == v1.Succeeded || vmi.Status.Phase == v1.Unknown || vmi.Status.Phase == v1.VmPhaseUnset {
+		writeError(errors.NewConflict(v1.Resource("virtualmachine"), name, fmt.Errorf("VM has status %q and is not running", vmi.Status.Phase)), response)
+		return
+	}
+
+	if vmi.Status.Phase == v1.Failed && bodyStruct.GracePeriod == nil {
+		writeError(errors.NewConflict(v1.Resource("virtualmachine"), name, fmt.Errorf("VM is Failed. Use `--force --gracePeriod=0` to stop the VM")), response)
```

Comment:
I believe the flag is `--grace-period` (and not `--gracePeriod`), isn't it?

Reply:
you are right. mixed up my flags and golang syntax. thank you!